### PR TITLE
[SW-216003] Enable regional compilation in text generation 

### PR DESCRIPTION
**Pull request by karol-brejna-i**
Enable regional compilation and force_static_compile in text generation workloads. This would help reduce compilation time and resolve OOM issue in Mixtral and Llama inference.

### CODE REVIEW

**Comment by yeonsily** [comment]
@karol-brejna-i Can you please give an example to use this regional compile? Also run CI with your change?

**Comment by kyotoyx** [comment]
> @karol-brejna-i Can you please give an example to use this regional compile?

Hi, here's an example of using regional compile in text generation task:
`
PT_HPU_LAZY_MODE=0 python3 ../gaudi_spawn.py --use_deepspeed --world_size 8 run_generation.py   --model_name_or_path /mnt/weka/data/pytorch/llama3.1/Meta-Llama-3.1-70B-Instruct/ --attn_softmax_bf16 --use_kv_cache --max_new_tokens 2048 --bf16 --batch_size 443 --use_flash_attention --flash_attention_recompute --bucket_size=128 --bucket_internal --trim_logits --max_input_tokens 2048 --warmup 2 --torch_compile --regional_compile --cache_size_limit 128
`

This workload will met OOM issue without "--regional_compile --cache_size_limit 128"

File: examples/text-generation/run_generation.py [hunk 1]
```diff
@@ -383,6 +383,29 @@ def setup_parser(parser):
         help="Specify the batch size split for attention and mlp layers. 1 for no split. This is enabled only for prompt.",
     )
 
+    parser.add_argument(
+        "--use_mark_dynamic",
```
[the remark this answers is not in the document]
Comment (regisss) [reply examples/text-generation/run_generation.py:387]: I agree with @yeonsily, this is not used anywhere. Please clarify why you're adding this arg.

**Comment by karol-brejna-i** [comment]
@yeonsily Do you have more comments?

**Comment by yeonsily** [comment]
@karol-brejna-i @kyotoyx Can you please remove --use_mark_dynamic if it's not used in this PR?
I was told that parameter is part of some other PR. Also I asked @kyotoyx  to update code as copilot review.

File: examples/text-generation/run_generation.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (karol-brejna-i) [reply examples/text-generation/run_generation.py]: Removed unused use_mark_dynamic argument.

**Comment by karol-brejna-i** [comment]
> @karol-brejna-i @kyotoyx Can you please remove --use_mark_dynamic if it's not used in this PR? I was told that parameter is part of some other PR. Also I asked @kyotoyx to update code as copilot review.
`use_mark_dynamic` removed.

**Review by regisss — APPROVED** [review]
